src/app/(dashboard)/list/announcements/page.tsx

```diff
@@ -26,42 +28,71 @@ const columns = [
   },
 ];
 
-type Announcement = {
-  id: number;
-  title: string;
-  class: string;
-  date: string;
-};
-const AnnouncementListPage = () => {
-  const renderRow = (item: Announcement) => (
-    <tr
-      key={item.id}
-      className="border-b border-gray-200 even:bg-slate-50 text-sm hover:bg-purple-50"
-    >
-      <td className="flex items-center gap-4 p-4">{item.title}</td>
-      <td>{item.class}</td>
-      <td className="hidden md:table-cell">{item.date}</td>
-      <td>
-        <div className="flex items-center gap-2">
-          {role === "admin" && (
-            //   <Link href={`/list/teachers/${item.id}`}>
-            //   <button className="h-7 w-7 rounded-full flex justify-center items-center bg-sky-200">
-            //     <Image src="/update.png" alt="" width={16} height={16} />
-            //   </button>
-            // </Link>
-            // <button className="h-7 w-7 rounded-full flex justify-center items-center bg-purple-200">
-            //   <Image src="/delete.png" alt="" width={16} height={16} />
-            // </button>
-            <>
-              <FormModal table="announcement" type="update" data={item} />
-              <FormModal table="announcement" type="delete" id={item.id} />
-            </>
-          )}
-        </div>
-      </td>
-    </tr>
-  );
+type AnnouncementList = Announcement & { class: Class }
+
+const renderRow = (item: AnnouncementList) => (
+  <tr
+    key={item.id}
+    className="border-b border-gray-200 even:bg-slate-50 text-sm hover:bg-purple-50"
+  >
+    <td className="flex items-center gap-4 p-4">{item.title}</td>
+    <td>{item.class.name}</td>
+    <td className="hidden md:table-cell">{new Intl.DateTimeFormat("en-US").format(item.date)}</td>
+    <td>
+      <div className="flex items-center gap-2">
+        {role === "admin" && (
+          //   <Link href={`/list/teachers/${item.id}`}>
+          //   <button className="h-7 w-7 rounded-full flex justify-center items-center bg-sky-200">
+          //     <Image src="/update.png" alt="" width={16} height={16} />
+          //   </button>
+          // </Link>
+          // <button className="h-7 w-7 rounded-full flex justify-center items-center bg-purple-200">
+          //   <Image src="/delete.png" alt="" width={16} height={16} />
+          // </button>
+          <>
+            <FormModal table="announcement" type="update" data={item} />
+            <FormModal table="announcement" type="delete" id={item.id} />
+          </>
+        )}
+      </div>
+    </td>
+  </tr>
+);
+const AnnouncementListPage = async ({
+  searchParams,
+}: {
+    searchParams: {
+    [key: string]: string | undefined
+  }
+}) => {
+
+  const { page, ...queryParams } = searchParams;
+  const p = page ? parseInt(page) : 1;
```

Comment:
never use single letter variables, it's extremely frowned upon and I was told by a technical recruiter that it's a quick sign to reject someone.

Single letter variables makes it hard to understand the context of a variable and you need to jump to its source to understand it.

I understand you needed to do this here because serachParams is returning a string (had the same issue as well last week), you can delay parsing the page (do it in pagination component) or rename it here like this
```suggestion
  const { page: rawPage, ...queryParams } = searchParams;
  const page = rawPage ? parseInt(rawPage) : 1;
```